src/main/scala/com/databricks/spark/xml/util/InferSchema.scala

```diff
@@ -104,6 +104,11 @@ private[xml] object InferSchema {
             Some(StructType(Seq(StructField(options.columnNameOfCorruptRecord, StringType))))
           case _: XMLStreamException =>
             None
+          case _: ClassCastException if shouldHandleCorruptRecord =>
```

Comment:
Could we do the similar thing with `NonFatal`?

Reply:
Fixed